flavorite_OAS_v1.yaml

```diff
@@ -0,0 +1,609 @@
+            application/json:
+              schema:
+                type: string
+        '400':
+          description: Invalid username/password supplied
+  /users/logout:
+    post:
+      tags:
+        - users
+      summary: Logs out current logged in user session
+      description: ''
+      operationId: logoutUser
+      parameters: []
+      responses:
+        '204':
+          description: successful operation
+  /users/{username}:
+    get:
+      tags:
+        - users
+      summary: Get user by user name
+      description: ''
+      operationId: getUserByName
+      parameters:
+        - name: username
+          in: path
+          description: 'The name that needs to be fetched. Use user1 for testing. '
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/User'
+        '400':
+          description: Invalid username supplied
+        '404':
+          description: User not found
+    put:
+      tags:
+        - users
+      summary: Update user
+      description: This can only be done by the logged in user.
+      operationId: updateUser
+      parameters:
+        - name: username
+          in: path
+          description: Name of user that need to be updated
+          required: true
+          schema:
+            type: string
+      requestBody:
+        $ref: '#/components/requestBodies/UpdateUser'
+      responses:
+        default:
+          description: successful operation
+    delete:
+      tags:
+        - users
+      summary: Delete user
+      description: This can only be done by the logged in user.
+      operationId: deleteUser
+      parameters:
+        - name: username
+          in: path
+          description: Name of user that needs to be deleted
+          required: true
+          schema:
+            type: string
+      responses:
+        '204':
+          description: Successful operation
+        '400':
+          description: Invalid username supplied
+        '404':
+          description: User not found
+  /users/{username}/reviews:
+    get:
+      tags:
+        - users
+      summary: Get reviews by user name
+      description: Returns all reviews by a single user
+      operationId: getReviewsByUsername
+      parameters:
+        - name: username
+          in: path
+          description: The name of user to return user's reviews
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: successful operation
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Review'
+        '400':
+          description: Invalid username supplied
+components:
+  schemas:
+    Review:
```

Comment:
This Review model also need required field. For example photoUrl could be empty when user did not upload the photo yet